automatic_cross_compile.sh

```diff
@@ -0,0 +1,61 @@
+# This script is intended to be used by the Jenkins night builds.
+# But it still possible to use manually setting up this variables:
+#
+# TARGET=[raspbian]
+# ROS2_DISTRO=[ardent,bouncy,crystal,master,...]
+
+# Check that all variables are defined before start
+if [[ -z "$TARGET" || -z "$ROS2_DISTRO" ]]
+then
+  echo "Error: environment variables are not defined!"
+  echo "Set values to: TARGET / ROS2_DISTRO"
+  exit 1
+fi
+
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+
+# Ensure that the cross-compilation environment is built and up-to-date
+BUILD_SCRIPT=$THIS_DIR/build.sh
+bash $BUILD_SCRIPT
+
+# Prepare cross-compilation environment
+SOURCE_SCRIPT=$THIS_DIR/env.sh
+source $SOURCE_SCRIPT $TARGET
+
+# Get sysroot
+SYSROOT_SCRIPT=$THIS_DIR/get_sysroot.sh
+bash $SYSROOT_SCRIPT
+
+# Remove ROS2 old cross-compilation workspace and get a new one
+CC_WS_DIR=~/ros2_cc_ws
+sudo rm -rf $CC_WS_DIR
+mkdir -p $CC_WS_DIR/src
+wget -O $CC_WS_DIR/ros2.repos https://raw.githubusercontent.com/ros2/ros2/"$ROS2_DISTRO"/ros2.repos
+vcs import $CC_WS_DIR/src < $CC_WS_DIR/ros2.repos
+echo "Created CC_WS_DIR=$CC_WS_DIR"
+
+# Get HEAD of branch
```

Comment:
I would update this to: Get first 7bytes of HEAD for the corresponding branch. Or something similar, but its also fine like this.